### PR TITLE
[PRM-827] - Added model for compliance view

### PR DESCRIPTION
**Pull request by LewisMitchell**

### CODE REVIEW

File: app/models/compliance/ObligationDetail.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+package models.compliance
+
+import play.api.libs.json.{Json, OFormat}
+
+case class ObligationDetail(
+                              status: String,
+                              inboundCorrespondenceFromDate: String,
+                              inboundCorrespondenceToDate: String,
+                              inboundCorrespondenceDateReceived: Option[String],
+                              inboundCorrespondenceDueDate: String,
```
Comment (AndrewRButcher) [inline app/models/compliance/ObligationDetail.scala:26]: Can we make these LocalDates?

File: app/models/compliance/ObligationDetail.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.compliance
+
+import play.api.libs.json.{Json, OFormat}
+
+case class ObligationDetail(
+                              status: String,
```
Comment (AndrewRButcher) [inline app/models/compliance/ObligationDetail.scala:22]: Can we create an Enum for the status?

**Review by AndrewRButcher — CHANGES_REQUESTED** [review]

File: app/models/compliance/ObligationDetail.scala [hunk 3]
```diff
@@ -18,13 +18,15 @@ package models.compliance
...
-                              inboundCorrespondenceToDate: String,
-                              inboundCorrespondenceDateReceived: Option[String],
-                              inboundCorrespondenceDueDate: String,
-                              periodKey: String
+                             status: ComplianceStatusEnum.Value,
+                             inboundCorrespondenceFromDate: LocalDate,
+                             inboundCorrespondenceToDate: LocalDate,
+                             inboundCorrespondenceDateReceived: Option[LocalDate],
+                             inboundCorrespondenceDueDate: LocalDate,
+                             periodKey: String
```
Comment (ruthdas03) [inline app/models/compliance/ObligationDetail.scala:29]: As per the schema , the property has a min and max length, which I guess needs to be checked.
                "periodKey":{
                           "type":"string",
                           "minLength":1,
                           "maxLength":4
                        }

File: app/models/compliance/ObligationIdentification.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.compliance
+
+import play.api.libs.json.{Json, OFormat}
+
+case class ObligationIdentification(
+                                   incomeSourceType: Option[String],
```
Comment (ruthdas03) [inline app/models/compliance/ObligationIdentification.scala:22]: The incomeSourceType has enum as per the schema and, under a given condition it becomes mandatory. Do we not consider these. 
"incomeSourceType":{
                        "type":"string",
                        "enum":[
                           "ITSA",
                           "ITSB",
                           "ITSP",
                           "ITSF"
                        ],
                        "description":"Mandatory if   ReferenceType  is an IncomeSourceID"
                     },

**Review by ruthdas03 — COMMENTED** [review]
There are couple of observations wrt the schema provided in confluence, you may want to consider.

File: app/models/compliance/ObligationDetail.scala (hunk 3 above)
Comment (LewisMitchell) [reply app/models/compliance/ObligationDetail.scala:29]: We don't really worry about the period key, I don't see it ever being used (it is purely for JSON parsing reasons)

File: app/models/compliance/ObligationIdentification.scala (hunk 4 above)
Comment (LewisMitchell) [reply app/models/compliance/ObligationIdentification.scala:22]: Reference type will never be IncomeSourceID until the service is made available under more regimes. So we don't consider this necessary (if we get the field, then fine - if not, we don't care)

**Review by ruthdas03 — APPROVED** [review]

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**